src/awkward/_v2/_connect/cling.py

```diff
@@ -54,11 +56,11 @@ def generate_ArrayView(compiler, use_cached=True):
   template <typename ARRAY, typename VALUE>
   class Iterator {
   public:
-    Iterator(ssize_t start, ssize_t stop, ssize_t which, ssize_t* ptrs)
-      : start_(start), stop_(stop), which_(which), ptrs_(ptrs) { }
+    Iterator(ssize_t start, ssize_t stop, ssize_t which, ssize_t* ptrs, PyObject* lookup = nullptr, PyObject* generator = nullptr)
```

Comment:
Since the `Iterators` are internally generated by us (not user-facing code), they don't have to have default arguments.

In fact, it's better if they don't because we would want to know immediately if we're ever creating an Iterator (or RIterator, or Array, or Record) that doesn't have these arguments set.

I can see how it's convenient to include a default value for a new argument while you're adding it, but eventually the default value should be taken off so that we can confirm that these objects are never getting `nullptr`s.

Reply:
done